vxyowsup/tests/test_whatsapp.py

```diff
@@ -120,6 +136,17 @@ def receive(self, data):
         '''
         self.toUpper(data)
 
+    def send_ack(self, node):
+        # TODO: rather send IncomingAckProtocolEntity and test extra fields
+        ack = AckProtocolEntity(_id=node['id'], _class='message')
+        self.receive(ack.toProtocolTreeNode())
+
+    def send_receipt(self, node):
+        # msg_class=None defualt indicates "delivered"
+        # alt: msg_class='read'
+        receipt = IncomingReceiptProtocolEntity(_id=node['id'], _from=node['to'], timestamp=node['id'].split('-')[0], type='read')
```

Comment:
Yeah, I think the timestamps just happened to be the same (or seem the same at a quick glance) as the message timestamp because they got sent so quickly. Maybe yowsup needed to set the timestamp manually for some internal reason.